Sources/NIOPosix/PosixSingletons.swift

```diff
@@ -0,0 +1,87 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+extension NIOGlobalSingletons {
+    /// A globally shared, lazily initialized ``MultiThreadedEventLoopGroup``  that uses `epoll`/`kqueue` as the
+    /// selector mechanism.
+    ///
+    /// The number of threads is determined by ``NIOGlobalSingletons/``
+    public static var posixEventLoopGroup: MultiThreadedEventLoopGroup {
+        return globalSingletonMTELG
+    }
+
+    /// A globally shared, lazily initialized ``NIOThreadPool`` that can be used for blocking I/O and other blocking operations.
+    ///
+    ///     /// The number of threads is determined by ``NIOSingletons.suggestedBlockingPoolThreadCount``.
+    public static var posixBlockingThreadPool: NIOThreadPool {
+        return globalPosixBlockingPool
+    }
+}
```

Comment:
Do we still see value in having this extension? Why would users choose to do

```
NIOGlobalSingletons.posixEventLoopGroup
```

over

```
MultiThreadedEventLoopGroup.globalSingleton
```

IMO we can just have the latter and we could rename the `NIOGlobalSingletons` to `NIOGlobalSingletonsConfiguration`

Reply:
I like having them in one place but of course it's not strictly necessary. I thought about renaming it to something like `Configuration` or `Suggestions` or so. But then I thought it's a good place for other extensions to hang their more specialised stuff off of.

For example, if we did change our mind and thought there should be a single threaded one too. Or maybe the first thread of the shared multi group is specially exposed as `NIOSingletons.posixSingleEventLoop` or something then we have a place in the API. If it's just "configuration" or "suggestions" or so, then that becomes awkard.

Reply:
tl;dr I think it gives us unexpensive/unoffensive design space. Users ain't gonna look there and we don't love singletons, let's give them their naughty corner but in there they have free reign

Reply:
Can we have an extension on `EventLoopGroup` à la https://github.com/apple/swift-nio/pull/2471#discussion_r1262318001 as well?

Reply:
@glbrntt Not quite sure what that buys us:

- For functions that take `MultiThreadedEventLoopGroup` today, that already works
- For functions that take `any EventLoopGroup`, we can only have _one_ `globalSingleton` and I'm not sure it's right to reserve that for `MultiThreadedEventLoopGroup`.

For the second point, let's say we have

```
extension EventLoopGroup where Self == MultiThreadedEventLoopGroup {
    public static var globalSingleton: MultiThreadedEventLoopGroup {
        return ...
    }
}

extension EventLoopGroup where Self == NIOTSEventLoopGroup {
    public static var globalSingleton: NIOTSEventLoopGroup {
        return ...
    }
}
```

if you were then to use `ClientBootstrap(group: .globalSingleton)` you'd get an ambiguity warning.

Reply:
It buys usability. Users don't need to know the type of the group, they can `.`-tab-complete for `any EventLoopGroup`. It's fine having to spell out the type on the `EventLoopGroup` extension: 

```swift
extension EventLoopGroup where Self == MultiThreadedEventLoopGroup {
    public static var globalMultiThreadedEventLoopGroup: MultiThreadedEventLoopGroup { // ... }
}

extension EventLoopGroup where Self == NIOTSEventLoopGroup {
    public static var globalNIOTSEventLoopGroup: NIOTSEventLoopGroup { // ... }
}
```

Reply:
AAhh, sorry, misread that. Yes, I'm happy if we use distinct names like `globalMultiThreadedEventLoopGRoup`.

Reply:
@glbrntt added